_posts/2022-04-25-tardigrade-launch.md

```diff
@@ -0,0 +1,84 @@
+
+Second you need to configure exchange manager. It can be done by creating the `etc/exchange-manager.properties` configuration file in the Trino working directory on coordinator and on each worker with the following configuration:
+  * `exchange-manager.name=filesystem`
+  * `exchange.base-directories=s3://<bucket-name>`
+  * `exchange.s3.region=us-east-1`
+  * `exchange.s3.aws-access-key=<access-key>`
+  * `exchange.s3.aws-secret-key=<secret-key>`
+
+And last you need to configure and enable task level retries:
+  * `retry-policy=TASK`
+  * `query.hash-partition-count=50` (more than 50 is currently not supported by the filesystem exchange implementation)
+
+It is also recommended to enable compression to reduce amount of data spooled on S3 (`exchange.compression-enabled=true`) as well as reduce the low memory killer delay to allow resource manager to unblock nodes running short on memory faster (`query.low-memory-killer.delay=0s`). Additionally we recommend enabling automatic writer scaling to optimize output file size for tables created with Trino (`scale-writers=true`) what could be critical for using Trino as an ETL engine.
+
+To increase overall throughput and reduce resource related task retries we recommend adjusting the concurrency settings based on the hardware configuration you have chosen. Here is the set of properties you need to set for the recommended hardware configuration (`32` vCPUs, `128GB` memory and `10Gbit/s` network):
+  * `task.concurrency=8`
+  * `task.writer-count=4`
+  * `fault-tolerant-execution-target-task-input-size=4GB`
+  * `fault-tolerant-execution-target-task-split-count=64`
+  * `fault-tolerant-execution-task-memory=5GB`
+
+## Potential issues
+
+* Queries may start failing with `software.amazon.awssdk.services.s3.model.S3Exception: Please reduce your request rate`:
+  * If your workload is I/O intensive it is possible that the S3 may start throttling some of the requests. The request limits are enforced per bucket. It is recommended to create an another bucket for spooling to allow Trino to balance the load `exchange.base-directories=s3://<first-bucket>,s3://<second-bucket>`
+* Queries may fail with `Task descriptor storage capacity has been exceeded`: 
+  * Trino has to maintain descriptors for each task in case it has to be restarted in an event of a failure. This information is currently stored in memory on coordinator. In the future we are planning to implement adaptive spilling for task descriptors, but at the momement there's a chance queries may hit this limit. If this happens we recommend:
+    * Run less queries concurrently
+    * Deploy a coordinator with more memory
+
+## Closing notes
+  
+Want to write about your experience using this, or become a community code contributor? Reach out to us about your submission, and we’ll send swag over as thank you for your submission.
+
+Want to get a free hosted Trino cluster to try things out? Reach out to Brian Zhan (brianjzhan@gmail.com).
```

Comment:
Suggest change to (to look more professional):

Want to get a free hosted Trino cluster to try things out? Fill out this form: https://forms.gle/6ac7csb9feMdD1mv9